packages/opentelemetry-core/test/context/B3Propagator.test.ts

```diff
@@ -98,127 +157,221 @@ describe('B3Propagator', () => {
+      assert.equal(context.getValue(PARENT_SPAN_ID_KEY), undefined);
     });
 
     it('should extract context of a sampled span from carrier when sampled is mentioned as boolean true flag', () => {
       carrier[X_B3_TRACE_ID] = '0af7651916cd43dd8448eb211c80319c';
       carrier[X_B3_SPAN_ID] = 'b7ad6b7169203331';
       carrier[X_B3_SAMPLED] = true;
-      const extractedSpanContext = getExtractedSpanContext(
-        b3Propagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
+      const context = b3Propagator.extract(
+        Context.ROOT_CONTEXT,
+        carrier,
+        defaultGetter
       );
+      const extractedSpanContext = getExtractedSpanContext(context);
 
       assert.deepStrictEqual(extractedSpanContext, {
         spanId: 'b7ad6b7169203331',
         traceId: '0af7651916cd43dd8448eb211c80319c',
         isRemote: true,
         traceFlags: TraceFlags.SAMPLED,
       });
+      assert.equal(context.getValue(DEBUG_FLAG_KEY), undefined);
+      assert.equal(context.getValue(PARENT_SPAN_ID_KEY), undefined);
     });
 
     it('should extract context of a sampled span from carrier when sampled is mentioned as boolean false flag', () => {
       carrier[X_B3_TRACE_ID] = '0af7651916cd43dd8448eb211c80319c';
       carrier[X_B3_SPAN_ID] = 'b7ad6b7169203331';
       carrier[X_B3_SAMPLED] = false;
-      const extractedSpanContext = getExtractedSpanContext(
-        b3Propagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
+      const context = b3Propagator.extract(
+        Context.ROOT_CONTEXT,
+        carrier,
+        defaultGetter
       );
+      const extractedSpanContext = getExtractedSpanContext(context);
 
       assert.deepStrictEqual(extractedSpanContext, {
         spanId: 'b7ad6b7169203331',
         traceId: '0af7651916cd43dd8448eb211c80319c',
         isRemote: true,
         traceFlags: TraceFlags.NONE,
       });
+      assert.equal(context.getValue(DEBUG_FLAG_KEY), undefined);
+      assert.equal(context.getValue(PARENT_SPAN_ID_KEY), undefined);
+    });
+
+    it('should extract context of a debug span from carrier when debug flag is valid', () => {
+      carrier[X_B3_TRACE_ID] = '0af7651916cd43dd8448eb211c80319c';
+      carrier[X_B3_SPAN_ID] = 'b7ad6b7169203331';
+      carrier[X_B3_FLAGS] = '1';
+      const context = b3Propagator.extract(
+        Context.ROOT_CONTEXT,
+        carrier,
+        defaultGetter
+      );
+      const extractedSpanContext = getExtractedSpanContext(context);
+
+      assert.deepStrictEqual(extractedSpanContext, {
+        spanId: 'b7ad6b7169203331',
+        traceId: '0af7651916cd43dd8448eb211c80319c',
+        isRemote: true,
+        traceFlags: TraceFlags.SAMPLED,
+      });
+      assert.strictEqual(context.getValue(DEBUG_FLAG_KEY), '1');
+      assert.equal(context.getValue(PARENT_SPAN_ID_KEY), undefined);
+    });
+
+    it('should extract context of a span from carrier when debug flag is invalidly set as 0', () => {
```

Comment:
Could you also add tests for flag for values other than 0 & 1?

Reply:
sure

Reply:
Like "True"/"False"/Some random number etc.